hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/filtermanagement/TargetFilterAddUpdateLayout.java

```diff
@@ -135,9 +151,33 @@ public ComponentContainer getRootComponent() {
         return filterQueryLayout;
     }
 
+    protected void setCSVExportCallback(final ProxyTargetFilterQuery proxyEntity){
+        StreamResource csvResource = getStreamResourceToCSV(proxyEntity);
+        fileDownloader = new FileDownloader(csvResource);
+        fileDownloader.extend(exportButton);
+    }
+
+    protected void removeDownloaderExtension(){
+        if(fileDownloader != null){
+            fileDownloader.remove();
+        }
+    }
+
+    private StreamResource getStreamResourceToCSV(final ProxyTargetFilterQuery proxyEntity){
+
+        StreamSource streamSource =  () -> {
```

Comment:
And start the processing in its own thread using the ui executor:
```suggestion
        final StreamSource streamSource = () -> {
            final PipedInputStream in = new PipedInputStream();

            try {
                final PipedOutputStream out = new PipedOutputStream(in);
                final TargetDataCsvExporter csvExporter = new TargetDataCsvExporter(targetManagement, ";",
                        "E MMM dd HH:mm:ss z yyyy");

                SpringContextHolder.getInstance().getBean("uiExecutor", Executor.class)
                        .execute(() -> csvExporter.writeTargetsByFilterIdTo(targetFilterQuery.getId(), out));
            } catch (final IOException ex) {
                LOG.warn("Creating piped Stream failed: ", ex);
            }

            return in;
        };

        return new StreamResource(streamSource, targetFilterQuery.getName() + ".csv");
```